_posts/pl/2021-02-10-przyklad-migracji-do-chmury.md

```diff
@@ -0,0 +1,888 @@
+
+- konfiguracji Cloud Build (**function/deploy.cloudbuild.yaml**):
+
+```yaml
+steps:
+  - id: fetch-dependencies-cache
+    name: gcr.io/cloud-builders/gsutil
+    entrypoint: bash
+    args: ['../yarn-cache-builder/yarn-cache-fetch.sh', '${_CACHE_BUCKET}']
+    dir: 'function'
+  - id: yarn-install
+    name: node:12.13.1
+    entrypoint: yarn
+    args: ['install']
+    dir: 'function'
+  - id: deploy-function
+    name: gcr.io/cloud-builders/gcloud
+    args: [
+      'beta', 'functions', 'deploy', 'demo-function',
+      '--runtime', 'nodejs10',
+      '--trigger-topic', '${_PUBSUB_TOPIC}',
+      '--region', '${_GCP_REGION}',
+      '--timeout', '10',
+      '--memory', '128MB',
+      '--max-instances', '1',
+      '--entry-point', 'main',
+      '--set-env-vars', 'COLLECTION_NAME=${_FIRESTORE_COLLECTION}'
+    ]
+    dir: 'function'
+  - id: push-dependencies-cache
+    name: gcr.io/cloud-builders/gsutil
+    entrypoint: bash
+    args: ['../yarn-cache-builder/yarn-cache-push.sh', '${_CACHE_BUCKET}']
+    dir: 'function'
+```
+
+Konfiguracja Cloud Build składa się z kolejnych kroków, których efektem będzie działająca funkcja, wyzwalana za pomocą Cloud Pub/Sub. Kroki związane z kopią podręczną zależności są analogiczne do konfiguracji aplikacji frontendowej. Ponieważ funkcja jest napisana w czystym JavaScript, to nie wymaga budowania z TypeSript do JavaScript (jak w przypadku aplikacji frontendowej), dlatego pominięty został krok wykonujący polecenie **yarn-build**. W linii 7 wykonywane jest jedynie polecenie **yarn-install**, które pobiera wymagane przez funkcję zależności. W kroku **deploy-function** tworzona jest funkcja na podstawie kodów źródłowych, zawartych w folderze **function**. Do konfiguracji zostały przekazane zmienne: **\${_PUBSUB_TOPIC}** (zawierająca nazwę topicu, który triggeruje funkcję), **\${_GCP_REGION}** (zawierająca nazwę regionu, w którym zostanie uruchomiona funkcja), **\${_FIRESTORE_COLLECTION}** (zawierająca nazwę kolekcji w Cloud Firestore, w której funkcja będzie utrwalać dane) oraz zmienna zawierająca nazwę bucketu przechowującego kopię podręczną zależności.
+
+Ponieważ triggery uruchamiające Cloud Build po zmianach konfiguracji infrastruktury (w folderze **infrastructure**) są tworzone przez ten sam kod, to za pierwszym razem trzeba utworzyć je ręcznie. W tym celu, w folderze **infrastructure** należy wykonać polecenia:
+
+- **terraform init** - inicjalizuje Terraform, tworząc folder **.terraform** zawierający stan i wykorzystywane wtyczki,
+- **terraform plan** - wykonuje plan działania Terraform, dzięki czemu można zweryfikować konfigurację pod kątem błędów, jeszcze przed uruchomieniem narzędzia,
+- **terraform apply** - akceptuje zmiany wynikające ze stanu aktualnego oraz wynikającego z konfiguracji, czyli wdraża infrastrukturę do chmury. W efekcie tego polecenia, utworzone zostaną wszystkie zasoby (triggery Cloud Build, buckety Cloud Storage, topic Cloud Pub/Sub).
```

Comment:
Zarówno tutaj, jak i w innych, podobnych przypadkach, rekomenduję zastosowanie średnika na końcu wyliczania, zamiast przecinków. Tutaj masz wyjaśnienie:
http://przepis-na-lo.pl/2012/09/jak-stworzyc-poprawne-wyliczenia/
(w skrócie - jeżeli wyliczenie zawiera przecinek, to powinno kończyć się średnikiem

Reply:
Czy mogę wszędzie użyć średników, czy tylko w zdaniach które zawierają przecinek, a w pozostałych przecinki?
Co w przypadku, kiedy jedno zdanie ma w sobie przecinek, a kolejne nie? Wtedy to jedno kończę średnikiem, kolejne przecinkiem, czy wszystkie średnikiem w obrębie tego wyliczenia?

Reply:
"Dopuszczalna jest przechodniość w górę tego schematu, tzn. zamiast przecinka można postawić średnik lub kropkę, zaś zamiast średnika — kropkę." - dam wszędzie średniki i kropkę na końcu.